src/util/gov.ts

```diff
@@ -53,112 +106,206 @@ export const decodeContent = (content?: Any): PropDecoded => {
+        typeUrl: url,
+        value: Carbon.Alliance.MsgUpdateAllianceProposal.decode(value),
       }
     case ProposalTypes.DeleteAlliance:
       return {
-        ...content,
-        value: Carbon.Alliance.MsgDeleteAllianceProposal.decode(content.value),
+        typeUrl: url,
+        value: Carbon.Alliance.MsgDeleteAllianceProposal.decode(value),
+      }
+    // update with new proposal messages
+    case ProposalTypes.SetMsgGasCostV2:
+      return {
+        typeUrl: url,
+        value: Carbon.Fee.MsgSetGasCost.decode(value),
+      }
+    case ProposalTypes.SetMinGasPriceV2:
+      return {
+        typeUrl: url,
+        value: Carbon.Fee.MsgSetMinGasPrice.decode(value),
+      }
+    case ProposalTypes.RemoveMsgGasCostV2:
+      return {
+        typeUrl: url,
+        value: Carbon.Fee.MsgRemoveGasCost.decode(value),
+      }
+    case ProposalTypes.RemoveMinGasPriceV2:
+        return {
+          typeUrl: url,
+          value: MsgRemoveMinGasPrice.decode(value),
+        }
+    case ProposalTypes.CommunityPoolSpendV2:
+      return {
+        typeUrl: url,
+        value: MsgCommunityPoolSpend.decode(value),
+      }
+    case ProposalTypes.SoftwareUpgradeV2:
+      return {
+        typeUrl: url,
+        value: MsgSoftwareUpgrade.decode(value),
+      }
+    case ProposalTypes.CancelSoftwareUpgradeV2:
+      return {
+        typeUrl: url,
+        value: MsgCancelUpgrade.decode(value),
       }
+    case ProposalTypes.CreateAllianceV2:
+        return {
+          typeUrl: url,
+          value: Carbon.Alliance.MsgCreateAlliance.decode(value),
+        }
+    case ProposalTypes.UpdateAllianceV2:
+        return {
+          typeUrl: url,
+          value: Carbon.Alliance.MsgUpdateAlliance.decode(value),
+        }
+    case ProposalTypes.DeleteAllianceV2:
+        return {
+          typeUrl: url,
+          value: Carbon.Alliance.MsgDeleteAlliance.decode(value),
+        }
+    case ProposalTypes.UpdateMarketV2:
+        return {
+          typeUrl: url,
+          value: Carbon.Market.MsgUpdateMarket.decode(value),
+        }
+    case ProposalTypes.UpdatePoolV2:
+        return {
+          typeUrl: url,
+          value: Carbon.Liquiditypool.MsgUpdatePool.decode(value),
+        }
+    case ProposalTypes.SetRewardCurveV2:
+        return {
+          typeUrl: url,
+          value: Carbon.Liquiditypool.MsgSetRewardCurve.decode(value),
+        }
+    case ProposalTypes.SetRewardsWeightsV2:
+        return {
+          typeUrl: url,
+          value: Carbon.Liquiditypool.MsgSetRewardsWeights.decode(value),
+        }
+    case ProposalTypes.SetCommitmentCurveV2:
+        return {
+          typeUrl: url,
+          value: Carbon.Liquiditypool.MsgSetCommitmentCurve.decode(value),
+        }
+    // case ProposalTypes.SettlementPriceV2:
+    //     return {
+    //       typeUrl: url,
+    //       value: Carbon.Pricing.MsgUpdateSettlementPrice.decode(value),
+    //     }
```

Comment:
Can remove if not in use.

Reply:
resolved in https://github.com/Switcheo/carbon-js-sdk/pull/440/commits/1d2943930756fb41647341556612ec627fc465c6